app/main.py

```diff
@@ -17,11 +17,16 @@ def create_person_list(people_list: list) -> list:
         wife_name = person_dict.get("wife")
         husband_name = person_dict.get("husband")
 
-        if wife_name and wife_name in Person.people:
-            person_instance.wife = Person.people[wife_name]
-            Person.people[wife_name].husband = person_instance
-        if husband_name and husband_name in Person.people:
-            person_instance.husband = Person.people[husband_name]
-            Person.people[husband_name].wife = person_instance
+        if wife_name:
+            wife_instance = Person.people.get(wife_name)
+            if wife_instance:
+                person_instance.wife = wife_instance
+                wife_instance.husband = person_instance
+
+        if husband_name:
+            husband_instance = Person.people.get(husband_name)
+            if husband_instance:
+                person_instance.husband = husband_instance
+                husband_instance.wife = person_instance
```

Comment:
Were no need to rewrite anything except of the conditions. 
`You can replace wife_name in Person.people with Person.people.get(wife_name) and it will give you the best performance - first operation has complexity of O(n) and second O(1) - best possible performance` -- return previous version and just change conditions from `if wife_name and wife_name in Person.people:` to `if Person.people.get(wife_name)`